WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit Grouped Products/GroupedProductListSelectorDataSource.swift

```diff
@@ -0,0 +1,112 @@
+
+/// Configures the results and cells for a paginated list of linked products of a grouped product, syncs each page of products,
+/// and handles actions that could alter the linked products.
+final class GroupedProductListSelectorDataSource: PaginatedListSelectorDataSource {
+    typealias StorageModel = StorageProduct
+
+    lazy var customResultsSortOrder: ((Product, Product) -> Bool)? = { [weak self] (lhs, rhs) in
+        guard let self = self else {
+            return true
+        }
+        let lhsProductID = lhs.productID
+        let rhsProductID = rhs.productID
+        let productIDs = self.groupedProductIDs
+        guard let lhsProductIDIndex = productIDs.firstIndex(of: lhsProductID), let rhsProductIDIndex = productIDs.firstIndex(of: rhsProductID) else {
+            return true
+        }
+        return lhsProductIDIndex < rhsProductIDIndex
+    }
+
+    // Observable list of the latest grouped product IDs
+    var productIDs: Observable<[Int64]> {
+        productIDsSubject
+    }
+    private let productIDsSubject: PublishSubject<[Int64]> = PublishSubject<[Int64]>()
+
+    private(set) var groupedProductIDs: [Int64] = [] {
+        didSet {
+            if groupedProductIDs != oldValue {
+                productIDsSubject.send(groupedProductIDs)
+            }
+        }
+    }
+
+    var selected: Product?
+
+    private let siteID: Int64
+    private let product: Product
+    private let imageService: ImageService
+
+    init(product: Product, imageService: ImageService = ServiceLocator.imageService) {
+        self.siteID = product.siteID
+        self.product = product
+        self.groupedProductIDs = product.groupedProducts
+        self.imageService = imageService
+    }
+
+    func createResultsController() -> ResultsController<StorageProduct> {
+        let storageManager = ServiceLocator.storageManager
+        let predicate = NSPredicate(format: "siteID == %lld AND productID IN %@", siteID, groupedProductIDs)
+        return ResultsController<StorageProduct>(storageManager: storageManager,
+                                                 matching: predicate,
+                                                 sortOrder: .nameAscending)
+    }
+
+    func handleSelectedChange(selected: Product) {
+        // no-op
+    }
+
+    func isSelected(model: Product) -> Bool {
+        return model == selected
+    }
+
+    func configureCell(cell: ProductsTabProductTableViewCell, model: Product) {
+        cell.selectionStyle = .none
+
+        let viewModel = ProductsTabProductViewModel(product: model)
+        cell.update(viewModel: viewModel, imageService: imageService)
+
+        let deleteButton = UIButton(type: .detailDisclosure)
+        deleteButton.setImage(.deleteCellImage, for: .normal)
+        deleteButton.tintColor = .systemColor(.tertiaryLabel)
+        deleteButton.on(.touchUpInside) { [weak self] _ in
+            self?.deleteProduct(model)
+        }
+        cell.accessoryView = deleteButton
```

Comment:
Maybe this can be a customization inside `ProductsTabProductTableViewCell`? Will we use this type of customization in other points of the app?

Reply:
I haven't seen other UI that has an accessory button for product cells, but it sounds better to move this to `ProductsTabProductTableViewCell`! updated in: https://github.com/woocommerce/woocommerce-ios/pull/2505/commits/95dc8e0b495dab86c297d5e0694496f4ccfa3c65